features/splash/src/main/java/com/kape/splash/ui/vm/SplashViewModel.kt

```diff
@@ -1,20 +1,42 @@
 package com.kape.splash.ui.vm
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.kape.httpclient.domain.UpdateClient
+import com.kape.router.EnterFlow
 import com.kape.router.ExitFlow
 import com.kape.router.Router
 import com.kape.vpnregions.utils.RegionListProvider
+import kotlinx.coroutines.launch
 import org.koin.core.component.KoinComponent
 
 class SplashViewModel(
     private val router: Router,
     private val regionListProvider: RegionListProvider,
+    private val updateClient: UpdateClient,
+    private val appUpdateUrl: String,
 ) : ViewModel(), KoinComponent {
 
+    private var updateUrl: String = ""
+
     fun load() {
         if (regionListProvider.isDefaultList()) {
             regionListProvider.loadVpnServerLatencies()
         }
-        router.handleFlow(ExitFlow.Splash)
+
+        viewModelScope.launch {
+            updateClient.invoke().collect {
```

Comment:
not that matters much at this point, but we are tying the splash to a network request which would bring back the slow start of the application on problematic networks.